src/Aspire.Hosting.MySql/MySqlBuilderExtensions.cs

```diff
@@ -25,7 +25,13 @@ public static class MySqlBuilderExtensions
     /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
     public static IResourceBuilder<MySqlServerResource> AddMySql(this IDistributedApplicationBuilder builder, string name, IResourceBuilder<ParameterResource>? password = null, int? port = null)
     {
-        var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password");
+        var passwordParameterName = $"{name}-password";
+        var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, passwordParameterName);
+
+        if (builder.ExecutionContext.IsRunMode && passwordParameter.Default is not null)
+        {
+            passwordParameter.Default = new UserSecretsParameterDefault(builder.AppHostAssembly, builder.Environment.ApplicationName, passwordParameterName, passwordParameter.Default);
+        }
```

Comment:
Would it make sense to extract this into a shared method, so it doesn't need to be copied everywhere?

Maybe we could just make `ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter` do this?

Reply:
I thought there was a reason I didn't do that but I'll take another look because I can't remember now :smile:

Reply:
Ah it was because all that API surface is public and based on `GenerateParameterDefault` rather than `ParameterDefault` so it wasn't immediately clean to do. I'll take another look.

Reply:
OK I moved this down to inside the call chain for `ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter`. Please take another look.

Reply:
Nice!